plutus-tx-plugin/test/Blueprint/Tests/Lib.hs

```diff
@@ -135,7 +137,36 @@ newtype Param2b = MkParam2b Bool
 $(PlutusTx.makeLift ''Param2b)
 $(makeIsDataSchemaIndexed ''Param2b [('MkParam2b, 0)])
 
-type Datum2 = Integer
+$( PlutusTx.asData
+    [d|
+      data Datum2 = MkDatum2
+        { datum2integer :: Integer
+        , datum2bool :: Bool
+        }
+        deriving stock (Generic)
+        deriving newtype (AsDefinitionId)
+      |]
+ )
+
+-- The TH splice above generates a 'Datum2' newtype wrapper around the 'BuiltinData'.
+-- Generically derived 'HasSchema Datum2' instance isn't useful as it describes such type
+-- as an opaque 'BuiltinData', so we provide a custom instance to show its structure.
```

Comment:
What if we exposed a version of the TH machinery that took a declaration quote (like `asData` does)? Then if you _have_ a declaration quote you could pass it to both, i.e.

```
let d = [d| ... |]

$(PlutusTx.asData d)
$(Blueprint.mkSchema d)
```
or something?

Reply:
Caveat: GHC stage restriction applies;